lib/ansible/modules/cloud/azure/azure_rm_snapshot.py

```diff
@@ -0,0 +1,378 @@
+                required=True
+            ),
+            sku=dict(
+                type='dict',
+                disposition='/',
+                options=dict(
+                    name=dict(
+                        type='str',
+                        choices=['Standard_LRS',
+                                 'Premium_LRS',
+                                 'Standard_ZRS']
+                    )
+                )
+            ),
+            os_type=dict(
+                type='str',
+                disposition='/properties/osType',
+                choices=['Windows',
+                         'Linux']
+            ),
+            creation_data=dict(
+                type='dict',
+                disposition='/properties/creationData',
+                required=True,
+                options=dict(
+                    create_option=dict(
+                        type='str',
+                        disposition='createOption',
+                        choices=['Empty',
+                                 'Attach',
+                                 'FromImage',
+                                 'Import',
+                                 'Copy',
+                                 'Restore',
+                                 'Upload'],
+                        required=True
+                    ),
+                    source_uri=dict(
+                        type='str',
+                        disposition='sourceUri'
+                    )
+                )
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.id = None
+        self.name = None
+        self.type = None
+        self.managed_by = None
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.url = None
+        self.status_code = [200, 201, 202]
+        self.to_do = Actions.NoAction
+
+        self.body = {}
+        self.query_parameters = {}
+        self.query_parameters['api-version'] = '2018-09-30'
+        self.header_parameters = {}
+        self.header_parameters['Content-Type'] = 'application/json; charset=utf-8'
+
+        super(AzureRMSnapshots, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                               supports_check_mode=True,
+                                               supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        for key in list(self.module_arg_spec.keys()):
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                self.body[key] = kwargs[key]
+
+        self.inflate_parameters(self.module_arg_spec, self.body, 0)
+
+        old_response = None
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(GenericRestClient,
```

Comment:
why use rest client? why not using sdk?

Reply:
it was faster to do without running into dependency problems.